diesel/src/pg/connection/mod.rs

```diff
@@ -169,6 +116,50 @@ impl GetPgMetadataCache for PgConnection {
     }
 }
 
+#[inline(always)]
+fn update_transaction_manager_status<T>(
+    query_result: QueryResult<T>,
+    conn: &mut PgConnection,
+) -> QueryResult<T> {
+    if let Err(Error::DatabaseError { .. }) = query_result {
+        /// avoid monomorphizing for every result type - this part will not be inlined
```

Comment:
This comment seems to disagree with the `#[inline(always)]` above.

Reply:
I want to always inline the above function, but not the `non_generic_inner` one

Reply:
I totally missed that inner function. Maybe explicitly put it outside the outer one to make this more clear?

Reply:
That doesn't seem worth the scoping pollution.
I think the "this part", the function name and call below, the fact it's a doc comment... should make enough clues for this to still be the best writing.